README.md

```diff
@@ -173,6 +183,73 @@ Kind: *Trusted Indication*.
 
 Errors:
 
+### `QueryPallet`
+
+Query the existence of a particular pallet type.
+
+- `name: Vec<u8>`: The name of the pallet to query.
+- `query_id`: The value to make the returned message identifiable with this query.
+
+Returns: A honestly populated `PalletInfo` to the origin.
+
+Safety: No concerns.
+
+Kind: *Instruction*
+
+Errors:
+
+### `PalletInfo`
+
+Provide information on the existence of a particular pallet type.
+
+- `query_id` The identifier of the query which caused this message to be sent.
+- `pallets: Vec<(u32, (u32, u32, u32))>`: The index/version of each of the pallets whose name matches the name in the according `QueryPallet` message.
+
+Safety: No concerns.
+
+Kind: *Trusted Indication*
+
+Errors:
+
+### `Dispatch`
+
+Dispatch a call into a pallet in the Frame system. This provides a means of ensuring that the pallet continues to exist with a known version.
+
+- `origin_type`: The means of expressing the message origin as a dispatch origin.
+- `name: Vec<u8>`: The name of the pallet to which to dispatch a message.
+- `major_minor: (u32, u32)`: The major and minor version of the pallet. The major version must be equal and the minor version of the pallet must be at least as great.
+- `pallet_index: u32`: The index of the pallet to be called.
```

Comment:
There's `name` and `pallet_index`. I think either can uniquely identify a remote pallet. Does it act as a double check?